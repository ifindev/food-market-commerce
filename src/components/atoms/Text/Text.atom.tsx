import clsx from 'clsx'
import { styled } from 'nativewind'
import { Text as NativeText } from 'react-native'

const baseHeading = clsx('text-black-darkish font-poppins-medium')
const HeadingOne = styled(NativeText, clsx(baseHeading, 'text-[32px]'))
const HeadingTwo = styled(NativeText, clsx(baseHeading, 'text-[22px]'))
const HeadingThree = styled(NativeText, clsx(baseHeading, 'text-xl'))

const baseBody = clsx('text-black-darkish font-poppins-regular')
const Body20 = styled(NativeText, clsx(baseBody, 'text-xl'))
const Body18 = styled(NativeText, clsx(baseBody, 'text-lg'))
const Body16 = styled(NativeText, clsx(baseBody, 'text-base'))
const Body14 = styled(NativeText, clsx(baseBody, 'text-sm'))
const Body13 = styled(NativeText, clsx(baseBody, 'text-[13px]'))
const Body12 = styled(NativeText, clsx(baseBody, 'text-xs'))

export default Object.assign(Body18, {
  HeadingOne,
  HeadingTwo,
  HeadingThree,
  Body20,
  Body16,
  Body14,
  Body13,
  Body12
})
