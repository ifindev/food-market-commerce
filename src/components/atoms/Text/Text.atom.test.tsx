import React from 'react'
import renderer from 'react-test-renderer'
import Text from './Text.atom'

describe('Text Atom Component', () => {
  it('should render HeadingOne correctly', () => {
    const tree = renderer.create(<Text.HeadingOne>Hello</Text.HeadingOne>)
    expect(tree).toMatchSnapshot()
  })

  it('should render HeadingTwo correctly', () => {
    const tree = renderer.create(<Text.HeadingTwo>Hello</Text.HeadingTwo>)
    expect(tree).toMatchSnapshot()
  })

  it('should render HeadingThree correctly', () => {
    const tree = renderer.create(<Text.HeadingTwo>Hello</Text.HeadingTwo>)
    expect(tree).toMatchSnapshot()
  })

  it('should render Body20 correctly', () => {
    const tree = renderer.create(<Text.Body20>Hello</Text.Body20>)
    expect(tree).toMatchSnapshot()
  })

  it('should render Body18 correctly', () => {
    const tree = renderer.create(<Text>Hello</Text>)
    expect(tree).toMatchSnapshot()
  })

  it('should render Body16 correctly', () => {
    const tree = renderer.create(<Text.Body16>Hello</Text.Body16>)
    expect(tree).toMatchSnapshot()
  })

  it('should render Body14 correctly', () => {
    const tree = renderer.create(<Text.Body14>Hello</Text.Body14>)
    expect(tree).toMatchSnapshot()
  })

  it('should render Body13 correctly', () => {
    const tree = renderer.create(<Text.Body13>Hello</Text.Body13>)
    expect(tree).toMatchSnapshot()
  })

  it('should render Body12 correctly', () => {
    const tree = renderer.create(<Text.Body13>Hello</Text.Body13>)
    expect(tree).toMatchSnapshot()
  })
})
