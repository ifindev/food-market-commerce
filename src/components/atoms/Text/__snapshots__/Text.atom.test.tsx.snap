// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Atom Component should render Body12 correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Regular",
      },
      {
        "fontSize": 13,
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text Atom Component should render Body13 correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Regular",
      },
      {
        "fontSize": 13,
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text Atom Component should render Body14 correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Regular",
      },
      {
        "fontSize": 14,
        "lineHeight": 20,
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text Atom Component should render Body16 correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Regular",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text Atom Component should render Body18 correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Regular",
      },
      {
        "fontSize": 18,
        "lineHeight": 28,
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text Atom Component should render Body20 correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Regular",
      },
      {
        "fontSize": 20,
        "lineHeight": 28,
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text Atom Component should render HeadingOne correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Medium",
      },
      {
        "fontSize": 32,
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text Atom Component should render HeadingThree correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Medium",
      },
      {
        "fontSize": 22,
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text Atom Component should render HeadingTwo correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#020202",
      },
      {
        "fontFamily": "Poppins-Medium",
      },
      {
        "fontSize": 22,
      },
    ]
  }
>
  Hello
</Text>
`;
